tools/toolchains/arm.py

```diff
@@ -554,7 +567,11 @@ def __init__(self, target, *args, **kwargs):
                     "ARM/ARMC6 compiler support is required for ARMC6 build"
                 )
 
-        if getattr(target, "default_toolchain", "ARMC6") == "uARM":
+        if  getattr(target, "default_lib", "std") == "small" and getattr(target, "microlib_supported", False) == False:    
```

Comment:
```suggestion
        if  target.default_lib == "small" and target.microlib_supported == False:    
```

Reply:
As explained in the previous comment we need getattr